### PR TITLE
Fix pylint errors due to version update

### LINKED ISSUE #1346: Night;y CI failing with latest pylint
pylint released a new version on Feb 25th and since then CI has been failing the nightly runs. Here is the log for the pylint checks

```
************* Module qiskit_nature.second_q.operators.spin_op
Warning: qiskit_nature/second_q/operators/spin_op.py:278:20: R1731: Consider using 'max_index = max(max_index, index)' instead of unnecessary if block (consider-using-max-builtin)
************* Module qiskit_nature.second_q.operators.fermionic_op
Warning: qiskit_nature/second_q/operators/fermionic_op.py:219:20: R1731: Consider using 'max_index = max(max_index, index)' instead of unnecessary if block (consider-using-max-builtin)
************* Module qiskit_nature.second_q.operators.bosonic_op
Warning: qiskit_nature/second_q/operators/bosonic_op.py:217:20: R1731: Consider using 'max_index = max(max_index, index)' instead of unnecessary if block (consider-using-max-builtin)
************* Module qiskit_nature.second_q.problems.properties_container
Warning: qiskit_nature/second_q/problems/properties_container.py:83:8: R1737: Use 'yield from' directly instead of yielding each element one by one (use-yield-from)

------------------------------------
Your code has been rated at 10.00/10

make: *** [Makefile:43: lint] Error 8
Error: Process completed with exit code 2.
```

### PR DESCRIPTION
**Pull request by MarcoBarroca**
### Summary
Fixes #1346

### Details and comments

Pylint had an update go through which made some new suggestions to the code. implemented those.

### CODE REVIEW

File: qiskit_nature/second_q/operators/bosonic_op.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2023.
+# (C) Copyright IBM 2024.
```
Comment (woodsp-ibm) [inline qiskit_nature/second_q/operators/bosonic_op.py:3]:
```suggestion
# (C) Copyright IBM 2023, 2024.
```

Comment (woodsp-ibm) [reply qiskit_nature/second_q/operators/bosonic_op.py:3]: This just one date when its just the one year, if it spans years then its the first year and last year.

**Comment by woodsp-ibm** [comment]
Thanks so much for splitting this out!

**Review by woodsp-ibm — APPROVED** [review]
LGTM, thx again!

**Comment by woodsp-ibm** [comment]
FYI if you run `make copyright` locally (from root in your cloned folder) it will fix the copyright dates as needed. Is the same tool as is used here that checks them.

**Comment by MarcoBarroca** [comment]
> Thanks so much for splitting this out!

No problem. I was just being lazy by repurposing the other PR

**Pull request merged**